### PR TITLE
Implement artifact download API endpoint

### PR DESCRIPTION
**Pull request by cutwater**
Closes: #1744

### CODE REVIEW

File: galaxy/api/v2/urls.py [hunk 1]
```diff
@@ -36,4 +38,7 @@
     path('collection-versions/<int:pk>/',
          views.CollectionVersionView.as_view(),
          name='collection-version-detail'),
+    path('collection-versions/<int:pk>/artifact/',
+         views.CollectionArtifactView.as_view(),
+         name='collection-version-artifact')
```
Comment (awcrosby) [inline galaxy/api/v2/urls.py]: Probably want the view `name` to be unique in case referring to them, such as with `reverse()`

**Comment by alikins** [comment]
In some quick browser tests:
GETs on http://localhost:8000/api/v2/collection-versions/16/artifact/
and http://localhost:8000/api/v2/collections/alikins/collection_ntp/versions/0.1.176/artifact/                                                                                

seems to work.

Using requests based 'http' seems to work as well
```
(galaxy-py36) [newswoop:F29:requests (master u=)]$ http -d --follow -p HBhb --verbose --all http://localhost:8000/api/v2/collections/alikins/collection_ntp/versions/0.1.176/artifact/
GET /api/v2/collections/alikins/collection_ntp/versions/0.1.176/artifact/ HTTP/1.1
Accept: */*
Accept-Encoding: identity
Connection: keep-alive
Host: localhost:8000
User-Agent: HTTPie/1.0.2



HTTP/1.1 302 Found
Access-Control-Allow-Origin: *
X-Powered-By: Express
allow: GET, HEAD, OPTIONS
connection: keep-alive
content-length: 0
content-type: text/html; charset=utf-8
date: Tue, 16 Apr 2019 16:52:36 GMT
location: /download/galaxy/alikins-collection_ntp-0.1.176.tar.gz
server: WSGIServer/0.2 CPython/3.6.6
vary: Accept, Cookie
x-frame-options: SAMEORIGIN



GET /download/galaxy/alikins-collection_ntp-0.1.176.tar.gz HTTP/1.1
Accept: */*
Accept-Encoding: identity
Connection: keep-alive
Host: localhost:8000
User-Agent: HTTPie/1.0.2



HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
X-Powered-By: Express
accept-ranges: bytes
connection: close
content-length: 7928
content-type: application/octet-stream
date: Tue, 16 Apr 2019 16:52:36 GMT
last-modified: Tue, 16 Apr 2019 15:20:29 GMT
server: Python/3.6 aiohttp/3.5.4

Downloading 7.74 kB to "alikins-collection_ntp-0.1.176.tar.gz"
Done. 7.74 kB in 0.00041s (18.67 MB/s)
```

File: galaxy/api/v2/urls.py (hunk 1 above)
Comment (cutwater) [reply galaxy/api/v2/urls.py]: According to django docs it's not necessary: https://docs.djangoproject.com/en/2.2/topics/http/urls/#naming-url-patterns

> You may also use the same name for multiple URL patterns if they differ in their arguments. In addition to the URL name, reverse() matches the number of arguments and the names of the keyword arguments.

**Comment by alikins** [comment]
I did a rudimentary merge of this branch and #1745 at https://github.com/ansible/galaxy/compare/devel...alikins:col_list_col_artifact

Using it with https://github.com/ansible/galaxy/issues/1742  seems to work

**Pull request merged**